File: SentiTrackAiFrontend/src/types/analytics.ts
export interface SentimentCounts {
  positive: number
  neutral: number
  negative: number
}

export interface DashboardAnalytics {
  total_entries: number
  analyzed_entries: number
  sentiment_counts: SentimentCounts
  average_confidence: number
  most_common_emotion: string | null
  most_common_mood: string | null
  current_streak: number
  longest_streak: number
  entries_this_week: number
  entries_this_month: number
  first_entry_at: string | null
  last_entry_at: string | null
}

export interface LabelCount {
  label: string
  count: number
}

export interface MoodDistribution {
  total_analyzed: number
  sentiment_counts: SentimentCounts
  emotions: LabelCount[]
  moods: LabelCount[]
}

export interface PeriodAnalytics {
  period: string
  entries: number
  analyzed: number
  sentiment_counts: SentimentCounts
  average_confidence: number
  most_common_emotion: string | null
}

export interface MonthlyAnalytics {
  year: number
  months: PeriodAnalytics[]
}

export interface YearlyAnalytics {
  years: PeriodAnalytics[]
}

File: SentiTrack/app/schemas/analytics.py
"""Analytics response schemas."""

from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, Field


class LabelCount(BaseModel):
    label: str
    count: int


class DashboardAnalytics(BaseModel):
    total_entries: int
    analyzed_entries: int
    sentiment_counts: dict[str, int] = Field(
        description="Counts keyed by positive | neutral | negative",
        examples=[{"positive": 5, "neutral": 2, "negative": 1}],
    )
    average_confidence: float | None
    most_common_emotion: str | None
    most_common_mood: str | None
    current_streak: int = Field(description="Consecutive days with an entry, ending today/yesterday")
    longest_streak: int = Field(description="Longest run of consecutive days with an entry")
    entries_this_week: int
    entries_this_month: int
    first_entry_at: datetime | None
    last_entry_at: datetime | None


class MoodDistribution(BaseModel):
    total_analyzed: int
    sentiment_counts: dict[str, int]
    emotions: list[LabelCount]
    moods: list[LabelCount]


class PeriodStat(BaseModel):
    period: str = Field(description="`YYYY-MM` for monthly, `YYYY` for yearly", examples=["2026-07"])
    entries: int
    analyzed: int
    sentiment_counts: dict[str, int]
    average_confidence: float | None
    most_common_emotion: str | None


class MonthlyAnalytics(BaseModel):
    year: int
    months: list[PeriodStat]


class YearlyAnalytics(BaseModel):
    years: list[PeriodStat]
